### ISSUE TITLE
Compiler fails silently if no output file is selected

### ISSUE THREAD

**Issue by mcantor** [guard/guard-coffeescript#22]
Create this folder structure:

```
.
|-- Gemfile
|-- Guardfile
|-- application.coffee
`-- static/
```

With this Guardfile:

```ruby
guard 'coffeescript', input: '', output: 'static' do
  watch 'application.coffee'
end
```

Guard will output `Successfully generated static/application.js` when `application.coffee` is changed, but no file will be created.

This is due to the `detect_nested_directories` method using `patterns.product(files)` to generate output locations when `:shallow` is set to `false` (the default).

Perhaps this error message would be appropriate:

```
ArgumentError: Ambiguous output directory for application.js. Try setting :shallow to true.
```

**Comment by e2** [comment]
Thanks. Could you create a PR for this?

**Comment by mcantor** [comment]
I'm not sure where the spec for it should go, nor exactly how to manipulate the inputs to faithfully recreate the error conditions. I'm assuming the integration test is the right place, but how much is being faked out? The test suite in general assumes non-`pwd` input directory.

**Comment by e2** [comment]
Yeah, it's not beautiful, but I'll be glad to help you as much as I can.

Since the fix is in the `detect_nested_directories` method I guess, it should be in the `runner_spec.rb`. It's about testing an edge case for a method, so don't worry about integration testing.

Maybe here: https://github.com/guard/guard-coffeescript/blob/5272ccd2134d3419f03b673ef540359ccee0a38c/spec/guard/coffeescript/runner_spec.rb#L122

The specs aren't great, so just copy a similar block, set the parameters to the ones you have, and just reproduce the problem for now.

If you're unsure about the setting up the conditions properly, just output values to the screen while having the problem, and check if you get the same values while running the spec.
